### PR TITLE
ONNX export test - minor fixes

### PR DESCRIPTION
**Pull request by galagam**
Edited: Minor fixes for ONNX export testing. 
* Consistently passing graphs' input names 
* Control whether to save input and output data with environment variable
* Minor refactor

### CODE REVIEW

**Comment by galagam** [comment]
@nzmora-nvidia fyi.

File: tests/pytorch/compare_fp8_accuracy_ort_trt.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# See LICENSE for license information.
+
+"""
+This script creates and compared results of ONNXRunTime and TensorRT on TransformerEngine models exported to ONNX.
```
Comment (ksivaman) [inline tests/pytorch/compare_fp8_accuracy_ort_trt.py:6]:
```suggestion
This script creates and compares results of ONNXRunTime and TensorRT on TransformerEngine models exported to ONNX.
```

File: tests/pytorch/compare_fp8_accuracy_ort_trt.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+The purpose of these tests is validation that TensorRT (TRT) achieves adequate accuracy on TE originated models.
+ONNXRunTime (ORT) is used here for reference. This script is meant to be used to obtain an accuracy status report
+and for debug purposes. It is not meant to be a part of TE CI at this point.
```
Comment (ksivaman) [inline tests/pytorch/compare_fp8_accuracy_ort_trt.py:10]:
```suggestion
The purpose of these tests to validate that TensorRT (TRT) achieves adequate accuracy on TE originated models.
ONNXRunTime (ORT) is used as a reference. This script is meant to be used to obtain an accuracy status report
and for debugging purposes. It is not meant to be a part of TE CI at this point.
```

File: tests/pytorch/test_onnx_export.py [hunk 3]
```diff
@@ -44,7 +44,9 @@
 # Global test configuration knobs.
 
 # Enable this to serialize test inputs and outputs to file (as a Polygraphy RunResults instance).
-SAVE_TEST_IO = False
+SAVE_TEST_IO = os.environ.get('SAVE_TEST_IO', None)
```
Comment (ksivaman) [inline tests/pytorch/test_onnx_export.py:47]:
```suggestion
SAVE_TEST_IO = os.environ.get('NVTE_SAVE_TEST_IO', None)
```

File: tests/pytorch/compare_fp8_accuracy_ort_trt.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+This script requires an input directory of ONNX models and their respective inputs and outputs.
+Either generate it by running: SAVE_TEST_IO=1 pytest test_onnx_export.py, or extract from
```
Comment (ksivaman) [inline tests/pytorch/compare_fp8_accuracy_ort_trt.py:13]:
```suggestion
Either generate it by running: NVTE_ONNX_EXPORT_SAVE_TEST_IO=1 pytest test_onnx_export.py, or extract from
```

**Comment by nzmora-nvidia** [comment]
LGTM

File: tests/pytorch/compare_fp8_accuracy_ort_trt.py (hunk 4 above)
Comment (galagam) [reply tests/pytorch/compare_fp8_accuracy_ort_trt.py]: File removed from PR

File: tests/pytorch/test_onnx_export.py (hunk 3 above)
Comment (galagam) [reply tests/pytorch/test_onnx_export.py]: Updated as part of the amended commit

File: tests/pytorch/compare_fp8_accuracy_ort_trt.py (hunk 2 above)
Comment (galagam) [reply tests/pytorch/compare_fp8_accuracy_ort_trt.py]: File removed from PR

File: tests/pytorch/compare_fp8_accuracy_ort_trt.py (hunk 1 above)
Comment (galagam) [reply tests/pytorch/compare_fp8_accuracy_ort_trt.py]: File removed from PR

File: tests/pytorch/test_onnx_export.py [hunk 5]
```diff
@@ -44,7 +44,9 @@
 # Global test configuration knobs.
 
 # Enable this to serialize test inputs and outputs to file (as a Polygraphy RunResults instance).
-SAVE_TEST_IO = False
+SAVE_TEST_IO = os.environ.get('NVTE_ONNX_EXPORT_SAVE_TEST_IO', None)
+SAVE_TEST_IO = True if SAVE_TEST_IO and SAVE_TEST_IO.lower() in ('1', 'true') else False
```
Comment (ksivaman) [inline tests/pytorch/test_onnx_export.py:48]:
```suggestion
SAVE_TEST_IO = bool(int(os.getenv("NVTE_ONNX_EXPORT_SAVE_TEST_IO", "0")))
```

Comment (ksivaman) [reply tests/pytorch/test_onnx_export.py:48]: Nitpick: simplification and clarity

**Review by ksivaman — APPROVED** [review]
LGTM.

**Pull request merged**